src/csp_iflist.c

```diff
@@ -154,8 +154,23 @@ csp_iface_t * csp_iflist_get_by_index(int idx) {
 
 int csp_iflist_add(csp_iface_t * ifc) {
 
+	/* Check if CSP is initialized, by allowing only loopback to be registered as initial interface */
+	extern csp_iface_t csp_if_lo;
+	if (interfaces == NULL && ifc != &csp_if_lo) {
+		return CSP_ERR_NOTSUP;
+	}
+
 	ifc->next = NULL;
 
+	/* Check for invalid config, allow using node 0 for loopback interface */
+	if ((interfaces != NULL && ifc->addr == 0) || ifc->addr > csp_id_get_max_nodeid()) {
```

Comment:
`ifc->addr > csp_id_get_max_nodeid()` allows assignment to the global broadcast address, which we may also want to deny with `ifc->addr >= csp_id_get_max_nodeid()`

Reply:
I also like the idea of checking `&csp_if_lo` instead of `ifc->addr == 0`. It's probably too late to implement node 0 as a valid address again, but not checking `ifc->addr == 0` would make us more compatible with old Satctl/modules.